ddmrp/models/stock_move.py

```diff
@@ -21,3 +21,48 @@ def _merge_moves_fields(self):
         res = super(StockMove, self)._merge_moves_fields()
         res["buffer_ids"] = [(4, m.id) for m in self.mapped("buffer_ids")]
         return res
+
+    def write(self, vals):
+        res = super(StockMove, self).write(vals)
+        if "state" in vals and self.env.company.ddmrp_auto_update_nfp:
+            self._update_ddmrp_nfp()
+        return res
+
+    def _update_ddmrp_nfp(self):
+        if self.env.context.get("no_ddmrp_auto_update_nfp"):
+            return True
+        buffer_model = self.env["stock.buffer"]
+        # Outgoing
+        out_buffers = self.env["stock.buffer"]
+        for move in self:
+            candidates = buffer_model.search(
+                [
+                    ("product_id", "=", move.product_id.id),
+                    ("location_id", "parent_of", move.location_id.id),
+                ]
+            )
+            for cand in candidates:
+                locations = self.env["stock.location"].search(
+                    [("id", "child_of", [cand.location_id.id])]
+                )
+                if move.location_dest_id not in locations:
+                    out_buffers += cand
+        # Incoming
+        in_buffers = self.env["stock.buffer"]
+        for move in self:
+            candidates = buffer_model.search(
+                [
+                    ("product_id", "=", move.product_id.id),
+                    ("location_id", "parent_of", move.location_dest_id.id),
+                ]
+            )
+            for cand in candidates:
+                locations = self.env["stock.location"].search(
+                    [("id", "child_of", [cand.location_id.id])]
+                )
+                if move.location_id not in locations:
+                    in_buffers += cand
```

Comment:
|= ?

Reply:
Not needed after following @guewen 's approach

Reply:
Thanks for reviewing! :smiley: